packages/react-components/src/shared/src/assertion.ts

```diff
@@ -3,6 +3,6 @@
 
 import { isNil } from "lodash";
 
-export function isNilOrEmpty(value) {
+export function isNilOrEmpty(value: any): value is null | undefined | "" {
```

Comment:
Is it a return type? `value is null | undefined | ""`

Reply:
it's a little bit different of a return type

![image](https://user-images.githubusercontent.com/38871812/107099179-f8d45e00-67de-11eb-94f5-307968e73c3d.png)

instead of saying the return value is a boolean, we let typescript know that if the return value is true, the value is either null, undefined or an empty string

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

the return value of this function is a boolean. however, we can let typescript know that if the return value is true, then value is either null, undefined or an empty string